lib/jquery.ui/jquery.ui.preview.js

```diff
@@ -48,8 +48,14 @@ $.widget( 'ui.preview', {
 	 * @protected
 	 */
 	_create: function() {
+		var hashBasedMessageProvider = new util.HashMessageProvider( this.options.messages );
 		if( this.options.messageProvider ) {
-			this.options.messageProvider.setDefaultMessages( this.options.messages );
+			this.options.messageProvider = new util.CombiningMessageProvider(
```

Comment:
Overwriting an option does not feel right as the user of the widget would receive a different object than the one set initially. Is that required for some reason or may the `messageProvider` be stored as direct member? And should these lines be moved to a `_setOption` reimplementation?